projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentService.java

```diff
@@ -146,7 +146,10 @@ public void updateDeployment(
           dockerRegistryService.dataJobImage(
               jobDeployment.getDataJobName(), jobDeployment.getGitCommitSha());
 
-      if (jobImageBuilder.buildImage(imageName, dataJob, jobDeployment, sendNotification)) {
+      String registrySecret = dockerRegistryService.getRegistrySecret();
+
+      if (jobImageBuilder.buildImage(
+          imageName, dataJob, jobDeployment, sendNotification, registrySecret)) {
```

Comment:
nit: there is one usage of the registry secret, so why creating explicitly a reference to it